server/ai_http.go

```diff
@@ -93,18 +94,35 @@ func aiHttpHandle[I any](h *lphttp, decoderFunc func(*I, *http.Request) error) h
 
 func (h *lphttp) StartLiveVideoToVideo() http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-
-		remoteAddr := getRemoteAddr(r)
-		ctx := clog.AddVal(r.Context(), clog.ClientIP, remoteAddr)
-
-		// skipping handleAIRequest for now until we have payments
-
 		var (
 			mid        = string(core.RandomManifestID())
 			pubUrl     = TrickleHTTPPath + mid
 			subUrl     = pubUrl + "-out"
 			controlUrl = pubUrl + "-control"
+
+			orch = h.orchestrator
 		)
+
+		remoteAddr := getRemoteAddr(r)
+		ctx := clog.AddVal(r.Context(), clog.ClientIP, remoteAddr)
+
+		// handle initial payment, the rest of the payments are done separately from the stream processing
+		payment, err := getPayment(r.Header.Get(paymentHeader))
+		if err != nil {
+			respondWithError(w, err.Error(), http.StatusPaymentRequired)
+			return
+		}
+		sender := getPaymentSender(payment)
+		_, ctx, err = verifySegCreds(ctx, h.orchestrator, r.Header.Get(segmentHeader), sender)
```

Comment:
Somewhere inside `verifySegCreds()` there is capacity check, something that will be related to the selection and session management (CC: @mjh1 ).